cabal.project

```diff
@@ -156,10 +156,10 @@ source-repository-package
 -- Direct dependency.
 source-repository-package
     type: git
-    location: https://github.com/Quviq/quickcheck-contractmodel
-    tag: cc43f13f98c704e0d53dbdef6a98367918f8c5c1
+    location: https://github.com/input-output-hk/quickcheck-contractmodel
+    tag: 071628000ad391ba6849b4d8407b5375228d235d
```

Comment:
I think you should be able to remove the `source-repository-package` for `quickcheck-contractmodel` and use the one provided in CHaP (https://github.com/input-output-hk/cardano-haskell-packages/blob/main/_sources/quickcheck-contractmodel/0.1.2/meta.toml)

Reply:
In theory yes, but in practise that means having to bump the CHaP index state and last time I tried that I got a whole bunch of version bounds issues.

Reply:
But let me try it and we see what happens. Worst case maybe I can get a second pair of eyes on this.

Reply:
@koslambrou and @zeme-iohk c.f. the following error I got out of left field after bumping the index state and removing the source repo stanza:
```
ⁱλ nix develop
warning: Git tree '/home/max/IOHK/plutus-apps' is dirty
error: builder for '/nix/store/mxqhn668ixhg3q67hw71smnnm2mgagmz-cardano-haskell-packages.drv' failed with exit code 1;
       last 1 log lines:
       > /nix/store/dfrz09vg1zigw8m5xwj90qn2si0736vy-source/root.json: openBinaryFile: does not exist (No such file or 
directory)
       For full logs, run 'nix log /nix/store/mxqhn668ixhg3q67hw71smnnm2mgagmz-cardano-haskell-packages.drv'.
error: 1 dependencies of derivation '/nix/store/cx7rmx3c96560qwyfwwd7hxvvmrpq544-dot-cabal.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nk1859zjpmh1nww1cqw1ydn0125s6gd5-haskell-project-plan-to-nix-pkgs.drv' 
failed to build
(use '--show-trace' to show detailed location information)
```
This is what's on this branch right now if you want to have a look.

Reply:
@andreabedini

Reply:
I am looking into this.

Reply:
I fixed a couple of small issues and added a constraint missing from ouroboros-consensus-shelley, see the diff here https://github.com/Quviq/plutus-apps/compare/PR-quickcheck-contractmodel-dep...andreabedini:plutus-apps:PR-quickcheck-contractmodel-dep
BUT I also ran in a bigger problem which is quickcheck-contractmodel ended up getting published to chap with the wrong version (index says it's 0.1.2 while cabal file says 0.1.2.0). This is definitely something we should have picked up in CI. I am coordinating a fix.

Reply:
For future reference, how did you debug such an error? Is there docs in CHaP which explain why we would get errors like this one? (`/nix/store/dfrz09vg1zigw8m5xwj90qn2si0736vy-source/root.json: openBinaryFile: does not exist (No such file or 
directory)`)

Reply:
@koslambrou `/nix/store/dfrz09vg1zigw8m5xwj90qn2si0736vy-source/root.json` not existing was telling me that the CHaP repository wasn't downloaded correctly (somehow). Then I looked into `/nix/store/dfrz09vg1zigw8m5xwj90qn2si0736vy-source` and I found there the wrong branch of the repo. See https://github.com/input-output-hk/plutus-apps/pull/970/files#r1121538038, tl;dr the CHaP repo has two branches, one for the metadata "sources" and one for the built repository.

Reply:
@andreabedini Thanks :)